portfolio/src/main/webapp/script.js

```diff
@@ -54,3 +54,26 @@ window.onload = () => {
     .then(text => { document.getElementById("post1").innerHTML = text; 
   });
 }
+
+/** 
+ * Fetch messages from server and insert them on the home page.
+ */
+window.onload = () => {
+  fetch('/data').then(response => response.json()).then(messages => {
+    const messagesListElement = document.getElementById('messages-container');
+    messagesListElement.innerHTML = '';
+    messagesListElement.appendChild(
+        createParagraphElement(messages[0]));
+    messagesListElement.appendChild(
+        createParagraphElement(messages[1]));
+    messagesListElement.appendChild(
+        createParagraphElement(messages[2]));
```

Comment:
not super important for this PR, just as an fyi.

you're making the assumption (correctly since you're hard-coding this in the Java servlet), that there are 3 items in the `messages` array. Note that if for any reason you change the Java but forget to change the Javascript for example that this might get out-of-sync.

For example, if the messages list becomes size 2 instead of size 3, then the messages[2] will (I think) become `undefined` in javascript. just keep in mind these assumptions as you code.

--

A cleaner way that will still work regardless of the messages size would just be to use a for loop on the messages size.

In pseudo-code:
````
for (message : messages) {
    messagesListElement.appendChild(createParagraphElement(message));
}
```

Reply:
made this edit!